### PR TITLE
Pokedex app 

### PR DESCRIPTION
**Pull request by jorge-prieto**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/user-event": "^12.8.3",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-redux": "^7.2.4",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.1.2",
+    "yarn": "^1.22.11"
```
Comment (hackeo1) [inline package.json]: don't add yarn as a dependency

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (hackeo1) [inline public/index.html]: change the title

File: src/components/Card.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import './Card.css'
+import { getIdFromPokenUrl } from '../helper/helps'
+
+export function Card({ name, url, onClick }) {
+  const image = 'https://raw.githubusercontent.com/PokeAPI/sprites/146c91287ad01f6e15315bbd733fd7442c91fe6d/sprites/pokemon/' + getIdFromPokenUrl(url) + '.png'
```
Comment (hackeo1) [inline src/components/Card.js]: use a central file to manage the base urls

File: src/components/Card.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  function pressed(e) {
+    e.preventDefault()
```
Comment (hackeo1) [inline src/components/Card.js]: don't use single char variables

File: src/components/Modal.css [hunk 5]
```diff
@@ -0,0 +1,55 @@
+.container {
+  display: flex;
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  z-index: 1;  
+  background-color: rgba(0,0,0, 0.3);
```
Comment (hackeo1) [inline src/components/Modal.css]: use variables for the colors

File: src/components/Modal.css [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  justify-content: center;
+  align-items: center;
+}
+.container .content {
+  display: block;
+  width: 60%;
+  height: 80%;
+  background-color: #eee;
```
Comment (hackeo1) [inline src/components/Modal.css]: use variables for the colors

File: src/components/Modal.css [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  border-radius: 12px;
+  box-shadow: 0px 0px 15px gray;
```
Comment (hackeo1) [inline src/components/Modal.css]: don't use named colors

File: src/components/Modal.css [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  padding: 20px;
+}
+.container .head {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  background-color: blue;
```
Comment (hackeo1) [inline src/components/Modal.css]: don't use named colors

File: src/components/Modal.css [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+}
+.container .between {
+  justify-content: space-between;
+}
+.container .center {
+  justify-content: center;
+  align-items: center;
+}
+.container .image {
+  float: left;
```
Comment (hackeo1) [inline src/components/Modal.css]: don't use float

File: src/components/Modal.css [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  object-fit: cover;
+}
+.container .wrap {
+  flex-flow: row wrap;
+}
+.container .btn {
+  margin: 5px;
+  padding: 5px;
+  background-color: #aaa;
```
Comment (hackeo1) [inline src/components/Modal.css]: use variables

File: src/components/Modal.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    <div className="container">
+      <div className="content flex column">
+        <div className="head">
+          <div className="flex row center">
+            <p className="title">{data?.name.toUpperCase()}</p>
+            <div className="btn">Compare To</div>
+          </div>
+          <a className="btn" onClick={onClose}>X</a>
+        </div>
+        <div className="">
```
Comment (hackeo1) [inline src/components/Modal.js]: remove unused attributes

File: src/components/Modal.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+            <tr>
+              <td>{data?.height}</td>
+              <td>{data?.weight}</td>
+              <td>{data?.gender === 0 ? 'Male' : 'Female'}</td>
+              <td><Ul data={data?.abilities} /></td>
+              <td><Ul data={data?.types} /></td>
+            </tr>
+          </table>
+        </div>
+        <div className="">FOOTER</div>
```
Comment (hackeo1) [inline src/components/Modal.js]: remove unused attributes

File: src/components/Modal.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+      </div>
+    </div>
+  ) : <></>
+}
+
+function useManager(url) {
+  const [data, setData] = useState()
+
+  useEffect(() => {
+    fetch(url)
```
Comment (hackeo1) [inline src/components/Modal.js]: fetch should be done using the store

File: src/components/NavBar.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+import './navbar.css'
+import LOGO from '../assets/pokemon-logo.png'
+import { appendFilter } from '../store/slicePokemons'
+
+export function NavBar() {
+  const [text, setText] = useState("")
+  const dispatch = useDispatch()
+
+  function onChange(e) {
```
Comment (hackeo1) [inline src/components/NavBar.js]: don't use single character variables

File: src/components/NavBar.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    setText(e.target.value)
+    dispatch(appendFilter(e.target.value))
+  }
+
+  return (
+    <div className="nav">
+      <div>
+          <img src={LOGO} alt='Pokemon logo' className='img-logo' />
```
Comment (hackeo1) [inline src/components/NavBar.js]: keep the consistency, double quotes here?

File: src/helper/helps.js [hunk 16]
```diff
@@ -0,0 +1,5 @@
+export const getIdFromPokenUrl = (url) => url.split('/')[6]
+
+export const getImage = (id) => {
+ return 'https://raw.githubusercontent.com/PokeAPI/sprites/146c91287ad01f6e15315bbd733fd7442c91fe6d/sprites/pokemon/' + id + '.png'
```
Comment (hackeo1) [inline src/helper/helps.js]: keep base urls centralized

File: src/pages/Home.css [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+}
+.img {
+    height: 40rem;
+}
+
+@media (max-width: 767px) {
+   .img {
+       height: 20rem;
+   }
+}
```
Comment (hackeo1) [inline src/pages/Home.css]: this file has different indentation

File: src/pages/PokemonsList.js [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+import './pokemonsList.css'
+
+export function PokemonsList() {
+  const { selected, data_filtered, onClickPokemons, onCloseModal} = useManager()
+
+  return (
+    <div>
+      <Modal visible={!!selected} url={selected} onClose={onCloseModal} />
+      <div className='grid-list'>{
+        data_filtered.map((el, i) => <Card key={i} name={el.name} url={el.url} onClick={onClickPokemons} />)
```
Comment (hackeo1) [inline src/pages/PokemonsList.js]: don't use single character variables

**Review by hackeo1 — CHANGES_REQUESTED** [review]
- naming convention for the assets must be consistent
- review the consistency between single a double quotes, when one is used and when the other

File: src/App.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
```
Comment (jugonz31) [inline src/App.js]: Add import for React

```suggestion
import React, { useEffect, useState } from 'react';
```

File: src/App.js [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+import { useSelector, useDispatch} from 'react-redux'
+import { MProvider} from './store/redux_store'
+import {fetchPokemons} from './store/slicePokemons'
+import { Navigator} from './components'
```
Comment (jugonz31) [inline src/App.js]: Try formatting the code, you need some extra spaces in all this: 
```suggestion
import { useSelector, useDispatch } from 'react-redux'
import { MProvider } from './store/redux_store'
import {fetchPokemons } from './store/slicePokemons'
import { Navigator } from './components'
```

Comment (jugonz31) [reply src/App.js]: I suggest you to check how to use 'Prettier' in your project, or you can just use the VSCode formatter

File: src/components/Card.css [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+}
+.card:hover {
+  transform: scale(1.1);
+}
+.card .card-image {
+  display: block;
+  width: 90%;
+  object-fit: contain;
+}
+.card .card-name {
```
Comment (jugonz31) [inline src/components/Card.css:19]: add spaces between class declarations: 
```suggestion
}

.card:hover {
  transform: scale(1.1);
}

.card .card-image {
  display: block;
  width: 90%;
  object-fit: contain;
}

.card .card-name {
```

File: src/components/Card.js [hunk 22]
```diff
@@ -0,0 +1,22 @@
+import { images } from '../Services';
```
Comment (jugonz31) [inline src/components/Card.js]: Add import statement for React

File: src/components/Modal.css [hunk 23]
```diff
@@ -0,0 +1,66 @@
+:root {
+  --off-white: #eee;
+  --gray: #909090;
+  --background: rgb(85, 214, 214);
+  --shadow: rgba(0, 0, 0, 0.3);
```
Comment (jugonz31) [inline src/components/Modal.css]: Would be better to define CSS variables in another place, so you can use them wherever you want and have it more organized

File: src/components/Modal.js [hunk 24]
```diff
@@ -0,0 +1,140 @@
+import { useState, useEffect } from 'react';
+import './Modal.css';
+import { getImage } from '../helper/helps';
+import { useDispatch, useSelector } from 'react-redux'
+import { fetchPoken, initCompare, clearModal } from '../store/sliceModal';
```
Comment (jugonz31) [inline src/components/Modal.js]: Import statements must have an order. Like this:
React imports should go first, followed by libraries and external tools (like react-redux in this case), followed by components and utils from your project and CSS files at the end.

There are tools that can do this for you. This is also part of formatting the code.

File: src/components/Modal.js [hunk 25]
```diff
@@ -0,0 +1,140 @@
...
+
+export function Modal({ visible, url, onClose }) {
+  const { left, right, is_compared, initialCompare, closeAndClean } = useManager(url, onClose);
```
Comment (jugonz31) [inline src/components/Modal.js]: always use lowerCamelCase to define constants, avoid using snake_case

File: src/components/Modal.js [hunk 26]
```diff
@@ -0,0 +1,140 @@
...
+
+  function closeAndClean() {
+    dispatch(clearModal())
+    onClose()
+  }
+
+  return { initialCompare, closeAndClean, is_compared, left, right };
+}
+
+function Ul({ data }) {
```
Comment (jugonz31) [inline src/components/Modal.js]: be careful and try naming functions in a better way 😅

File: src/components/Modal.js [hunk 27]
```diff
@@ -0,0 +1,140 @@
...
+  return (
+    <ul>
+      {data?.map((element, index) => (
+        <li key={index}>{element}</li>
+      ))}
+    </ul>
+  );
+}
+
+function PokenDescription({id, desc, height, weight, gender, abilities, types}) {
```
Comment (jugonz31) [inline src/components/Modal.js]: also naming constants or props
```suggestion
function PokenDescription({id, description, height, weight, gender, abilities, types}) {
```

File: src/components/Modal.js [hunk 28]
```diff
@@ -0,0 +1,140 @@
...
+          <td>
+            <Ul data={types} />
+          </td>
+        </tr>
+      </table>
+    </div>
+  )
+}
+
+function Compared({ left, right }) {
```
Comment (jugonz31) [inline src/components/Modal.js]: functions should never start with uppercase. always should be lowerCamelCase

File: src/components/Modal.js [hunk 29]
```diff
@@ -0,0 +1,140 @@
...
+                  }
+                </ul>
+              }
+              <p>{camp}</p>
+              {
+                camp !== 'Abilities' ? 
+                <p>{right[fields[camp]]}</p> : 
+                <ul>
+                  {
+                    right[fields[camp]].map((abilitie, key) => (
```
Comment (jugonz31) [inline src/components/Modal.js]: fix typo 'abilitie'

File: src/components/NavBar.js [hunk 30]
```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import './navbar.css';
+import LOGO from '../assets/pokemon-logo.png';
+import { appendFilter } from '../store/slicePokemons';
```
Comment (jugonz31) [inline src/components/NavBar.js]: this is also a valid way to order imports, it is acceptable

File: src/components/index.js [hunk 31]
```diff
@@ -0,0 +1,2 @@
+export * from './Navigator'
+export * from './Card'
```
Comment (jugonz31) [inline src/components/index.js]: add eof new line

File: src/helper/helps.js [hunk 32]
```diff
@@ -0,0 +1,7 @@
+import { images } from "../Services"
+
+export const getIdFromPokenUrl = (url) => url.split('/')[6]
+
+export const getImage = (id) => {
+ return images + id + '.png'
+}
```
Comment (jugonz31) [inline src/helper/helps.js]: add eof new line

File: src/pages/PokemonsList.js [hunk 33]
```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchPokemons } from '../store/slicePokemons';
+import { Card } from '../components';
+import { getIdFromPokenUrl } from '../helper/helps';
+import { Modal } from '../components/Modal';
+import './pokemonsList.css';
+
+export function PokemonsList() {
+  const { selected, data_filtered, onClickPokemons, onCloseModal } =
```
Comment (jugonz31) [inline src/pages/PokemonsList.js]: same issue with the snake_case

Comment (jugonz31) [reply src/pages/PokemonsList.js]: also fix the typo 'data filtered'

File: src/pages/PokemonsList.js [hunk 34]
```diff
@@ -0,0 +1,67 @@
...
+        ))}
+      </div>
+    </div>
+  );
+}
+
+const search = (array, text) =>
+  array.filter((element) => !element.name.toLowerCase().indexOf(text.toLowerCase()));
+
+function useManager() {
```
Comment (jugonz31) [inline src/pages/PokemonsList.js]: try always using arrow functions, it is more common

File: src/pages/PokemonsList.js [hunk 35]
```diff
@@ -0,0 +1,67 @@
...
+  const [selected, setSelect] = useState();
```
Comment (jugonz31) [inline src/pages/PokemonsList.js]:
```suggestion
  const [selected, setSelected] = useState();
```

File: src/pages/PokemonsList.js [hunk 36]
```diff
@@ -0,0 +1,67 @@
...
+  const dispatch = useDispatch();
+  const { array, textfilter } = useSelector((state) => state.pokemons);
```
Comment (jugonz31) [inline src/pages/PokemonsList.js]: lowerCamelCase for constants!!!
```suggestion
  const { array, textFilter } = useSelector((state) => state.pokemons);
```

File: src/pages/index.js [hunk 37]
```diff
@@ -0,0 +1,2 @@
+export * from './Home'
+export * from './PokemonsList'
```
Comment (jugonz31) [inline src/pages/index.js]: eof

File: src/pages/pokemonsList.css [hunk 38]
```diff
@@ -0,0 +1,6 @@
+.grid-list {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+  grid-gap: 2em;
+  padding: 2em;
+}
```
Comment (jugonz31) [inline src/pages/pokemonsList.css]: eof

**Review by jugonz31 — COMMENTED** [review]
- Add React import statement in all components
- please take a look at how you name items, classes, components, and folders in all the project:

> use UpperCamelCase for components (MyComponent.js - export default MyComponent)
> use camelCase for constants, props, variables, etc. (let myVariable = 0; const myConst = true;)
> use snake_case or kebab-case for CSS classes: (.my-class {})
> use kebab-case for folders (src/components/my-component/MyComponent.js)

- Try fixing import order and run a formatter in all your project 
- Use clearer names for everything

**Review by jugonz31 — APPROVED** [review]

File: src/components/Modal.js [hunk 39]
```diff
@@ -0,0 +1,107 @@
...
+  weight,
+  gender,
+  abilities,
+  types,
+}) {
+  return (
+    <div>
+      <img src={getImage(id)} className='image' />
+      <p>{desc}</p>
+      <table>
```
Comment (hackeo1) [inline src/components/Modal.js]: don't use table, use display table instead

File: src/store/sliceModal.js [hunk 40]
```diff
@@ -0,0 +1,68 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+    left: null,
+    right: null,
+    isCompared: false,
```
Comment (hackeo1) [inline src/store/sliceModal.js]: this file has 4 spaces for indentation instead of 2

**Review by hackeo1 — CHANGES_REQUESTED** [review]

**Review by hackeo1 — APPROVED** [review]

**Pull request merged**